specification/iotcentral/data-plane/Microsoft.IoTCentral/stable/2022-05-31/iotcentral.json

```
@@ -0,0 +1,3952 @@
+    },
+    "ComponentName": {
+      "in": "path",
+      "name": "componentName",
+      "description": "Name of the device component.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceGroupId": {
+      "in": "path",
+      "name": "deviceGroupId",
+      "description": "Unique ID for the device group.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "maxLength": 255,
+      "pattern": "^[a-zA-Z0-9_-]*$",
+      "required": true
+    },
+    "DeviceId": {
+      "in": "path",
+      "name": "deviceId",
+      "description": "Unique ID of the device.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "DeviceTemplateId": {
+      "in": "path",
+      "name": "deviceTemplateId",
+      "description": "Unique [Digital Twin Model Identifier](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#digital-twin-model-identifier) of the device template.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "ModuleName": {
+      "in": "path",
+      "name": "moduleName",
+      "description": "Name of the device module.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "OrganizationId": {
+      "in": "path",
+      "name": "organizationId",
+      "description": "Unique ID of the organization.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "maxLength": 48,
+      "pattern": "^(?!-)[a-z0-9-]{1,48}[a-z0-9]$",
+      "required": true
+    },
+    "Subdomain": {
+      "in": "path",
+      "name": "subdomain",
+      "description": "The application subdomain.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "required": true
+    },
+    "TelemetryName": {
+      "in": "path",
+      "name": "telemetryName",
+      "description": "Name of this device telemetry.",
+      "x-ms-parameter-location": "method",
+      "type": "string",
+      "required": true
+    },
+    "UserId": {
+      "in": "path",
+      "name": "userId",
+      "description": "Unique ID of the user.",
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "maxLength": 48,
+      "pattern": "^[a-zA-Z0-9-_]*$",
+      "required": true
+    },
+    "Version": {
+      "in": "query",
+      "name": "api-version",
+      "description": "The version of the API being called.",
+      "x-ms-parameter-location": "client",
+      "type": "string",
+      "required": true
+    }
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  }
```

You need to use `security` to specify "azure_auth" as the security requirement for the operations in this service.
```suggestion
  },
  "security" : [
    {
      "azure_auth": [ "user_impersonation" ]
    }
  }
```